### PR TITLE
customise the nav based on roles | Adniya

### PR DESCRIPTION
**Pull request by adniyaYousaf**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -10,24 +10,32 @@ import TendersList from "./TenderList";
...
-			<Route path="/buyer-dashboard" element={<BuyerDashboard />} />
-			<Route path="/list-tenders" element={<TendersList />} />
-			<Route path="/list-tenders/page/:pageNumber" element={<TendersList />} />
-			<Route path="/login" element={<LogInForm />} />
-		</Routes>
-		<Footer className="footer" />
-	</>
-);
+const App = () => {
+	const [role, setRole] = useState(null);
```
Comment (soonick) [inline client/src/App.js]: This is not needed here. We need to read the role from local storage when the header is rendered

File: client/src/pages/LogInForm.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ function LogInForm() {
 
 			localStorage.setItem("authToken", token);
 			localStorage.setItem("userType", user_type);
+			setRole(user_type);
```
Comment (soonick) [inline client/src/pages/LogInForm.js]: This is not the right way to do this. Returning users have a role without having to log in again.

Revert changes to this file. They are not needed.

File: client/src/Header.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import Logo from "./assets/images/CTY-logo-rectangle.png";
 import { NavLink } from "react-router-dom";
 import "./Header.css";
 
-const Header = () => {
+const Header = ({ role }) => {
```
Comment (soonick) [inline client/src/Header.js]: Read the role from localStorage on each render

File: client/src/Header.js [hunk 4]
```diff
@@ -22,6 +22,83 @@ const Header = () => {
...
+
+				{role === "admin" && (
+					<>
+						<NavLink
+							exact
+							to="/list-tenders"
+							className="nav-link"
+							activeClassName="active"
+						>
+							All Tenders{" "}
```
Comment (soonick) [inline client/src/Header.js]: Why not just `All Tenders`?

File: client/src/Header.js [hunk 5]
```diff
@@ -22,6 +22,83 @@ const Header = () => {
 						All Tenders{" "}
 					</NavLink>
```
Comment (soonick) [inline client/src/Header.js]: Remove this link and the `home` link

**Review by soonick — CHANGES_REQUESTED** [review]

File: client/src/Header.js (hunk 4 above)
Comment (adniyaYousaf) [reply client/src/Header.js]: I have removed it now.

File: client/src/Header.js [hunk 6]
```diff
@@ -1,27 +1,96 @@
...
+					<>
+						<NavLink
+							exact
+							to="/list-tenders"
+							className="nav-link"
+							activeClassName="active"
+						>
+							All Tenders
+						</NavLink>
+						<>
```
Comment (soonick) [inline client/src/Header.js]: Why do we have an extra level of indentation in there an in other places?

Comment (adniyaYousaf) [reply client/src/Header.js]: I have removed it.

**Review by soonick — APPROVED** [review]

**Pull request merged**